x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/TransportPutAutoFollowPatternAction.java

```diff
@@ -142,10 +142,22 @@ static ClusterState innerPut(PutAutoFollowPatternAction.Request request,
         followedLeaderIndices.put(request.getName(), followedIndexUUIDs);
         // Mark existing leader indices as already auto followed:
         if (previousPattern != null) {
-            markExistingIndicesAsAutoFollowedForNewPatterns(request.getLeaderIndexPatterns(), remoteClusterState.metadata(),
-                previousPattern, followedIndexUUIDs);
+            if (request.getLeaderIndexExclusionPatterns().equals(previousPattern.getLeaderIndexExclusionPatterns()) == false) {
+                throw new IllegalArgumentException("Updating the leader index exclusion patterns is not allowed");
+            }
```

Comment:
Is there a reason for this? I can imagine someone having a pattern and then they discover it matching something they did not want and want to update it to have an exclusion.

Reply:
My concern was that it's possible that an index that's already followed can be excluded when the auto follow patterns are updated, but we can document that if you're interested in unfollowing those indices you should do it manually after updating the patterns. For now, I've added a warning when a new exclusion pattern excludes an already followed index.